### DISCUSSION TITLE
Does (*UpdateQuery).Set() invalidate (*UpdateQuery).Column() ?

### DISCUSSION THREAD

**Discussion by nanmu42** [uptrace/bun#647] [Q&A] [answered]
Hi,

Due to https://github.com/uptrace/bun/discussions/621, I plan to manually update `updated_at`:

```go
type OAuth2State struct {
	ID string `json:"id" bun:",pk"`

	CreatedAt time.Time `bun:",nullzero,notnull,skipupdate,default:current_timestamp" json:"createdAt"`
	UpdatedAt time.Time `bun:",nullzero,notnull,default:current_timestamp" json:"updatedAt"`

	Mode             OAuth2StateMode `json:"mode" bun:",notnull,nullzero"`
	Status           OAuth2Status    `json:"status" bun:",notnull"`
}

func (op Operator) UpdateOAuth2StateStatusByID(ctx context.Context, id string, status OAuth2Status) (err error) {
	state := OAuth2State{ID: id, Status: status}
	err = assertAffectedOneRow(op.db.NewUpdate().
		Model(&state).
		Column("status").
		Set("updated_at = current_timestamp").
		WherePK().
		Exec(ctx))
	return
}
```

But I found out the generated SQL only sets `updated_at` and `status` is ommited.

I digged into the source and got a little suprised:

https://github.com/uptrace/bun/blob/efe03f464902f44e95c34c00c5feaacd5479f080/query_update.go#L243-L263

So using `Set()` results all `Column()` operation invalid? Is this by design?

Thanks and have a pleasant day!

**Comment by vmihailenco** [comment, marked as answer]
This is expected, but we could try to change this behavior if there is enough interest.
